flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorTest.java

```diff
@@ -120,13 +122,21 @@
 	private static final String TASK_MANAGER_LOCATION_INFO = "Unknown location";
 
 	private CheckpointFailureManager failureManager;
+	private Throwable failureCause;
+	private int failureCallbackInvocationCounter = 0;
 
 	@Rule
 	public TemporaryFolder tmpFolder = new TemporaryFolder();
 
 	@Before
 	public void setUp() throws Exception {
-		failureManager = new CheckpointFailureManager(0, () -> {});
+		failureCallbackInvocationCounter = 0;
```

Comment:
This new code is specific to only one test, I would suggest to create new `CheckpointFailureManager` inside the test to isolate those changes in that test alone.